bigquery/google/cloud/bigquery/_helpers.py

```diff
@@ -241,3 +241,66 @@ def _build_udf_resources(resources):
         udf = {resource.udf_type: resource.value}
         udfs.append(udf)
     return udfs
+
+
+class ScalarQueryParameter(object):
+    """Named / positional query parameters for scalar values.
+
+    :type name: str or None
+    :param name: Parameter name, used via `@foo` syntax.  If None, the
+                 paramter can only be addressed via position (`?`).
+
+    :type type_: str
+    :param type_: name of parameter type.  One of `'STRING'`, `'INT64'`,
+                  `'FLOAT64'`, `'BOOLEAN'`, `'TIMESTAMP'`, or `'DATE'`.
+
+    :type value: str, int, float, bool, :class:`datetime.datetime`, or
+                 :class:`datetime.date`.
+    :param value: the scalar parameter value.
+    """
+    def __init__(self, name, type_, value):
```

Comment:
I'd rather have the named form be "normal", and have the `positional` factory mark the other usage.